crates/transaction-pool/src/traits.rs

```diff
@@ -250,7 +250,7 @@ impl<T> BestTransactions for std::iter::Empty<T> {
 }
 
 /// Trait for transaction types used inside the pool
-pub trait PoolTransaction: fmt::Debug + Send + Sync + FromRecoveredTransaction {
+pub trait PoolTransaction: fmt::Debug + Send + Sync + Clone + FromRecoveredTransaction {
```

Comment:
why do we need Clone now?
this should not used because all of them are supposed to be Arc'ed